# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/svg/types/scripted/SVGLength-rem.html
# Commit: 2f48a6566514
# Full Hash: 2f48a65665147ef98c0f1e7d26a99f1931f9470d
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 21:17:53
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 1 - support rem, ch, ic and cap units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180618
# ==============================================================================

diff -r fc9333c3b9f7 -r 2f48a6566514 testing/web-platform/tests/svg/types/scripted/SVGLength-rem.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/svg/types/scripted/SVGLength-rem.html	Tue Sep 12 11:45:26 2023 +0000
@@ -0,0 +1,30 @@
+<!DOCTYPE HTML>
+<title>SVGLength with 'rem' unit</title>
+<link rel="help" href="https://www.w3.org/TR/SVG/types.html#InterfaceSVGLength">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<style>
+  :root {
+     /* default is 16px in all desktop browsers */
+     /* the math for 100px is: 16px * 0.25 = 4px/rem -> 100px / 4px/rem = 25rem = 100px */
+    font-size: 25%;
+  }
+</style>
+<div id="rem_ref" style="width:25rem"></div>
+<svg>
+  <rect id="rect" x="25rem"/>
+</svg>
+<script>
+  let ref_width = document.getElementById("rem_ref").offsetWidth;
+  let rem_length = document.getElementById("rect").x.baseVal;
+
+  test(() => {
+    assert_equals(rem_length.unitType, SVGLength.SVG_LENGTHTYPE_UNKNOWN);
+    assert_equals(rem_length.value, 100);
+  }, "rem unit in SVGLength");
+
+  test(() => {
+    rem_length.value = ref_width * 2;
+    assert_equals(rem_length.valueInSpecifiedUnits, 50);
+  }, "Convert back to rem from new user unit value");
+</script>
